### ISSUE TITLE
keyword-spacing not working at all

### ISSUE THREAD

**Issue by arimourao** [eslint/eslint#9243]
**Tell us about your environment**

* **ESLint Version:*4.6.1*
* **Node Version:*8.4*
* **npm Version:*5.4*

**What parser (default, Babel-ESLint, etc.) are you using?**
default
**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
"keyword-spacing": ["error", {
            "after": false
        }],


```

</details>

**What did you do? Please include the actual source code causing the issue.**

```js





```

**What did you expect to happen?**
I wanted to enforce the rule to add a space before the curly braces (i.e function() {} instead of function(){} )
**What actually happened? Please include the actual, raw output from ESLint.**
Nothing. Eslint is not reporting the error nor auto-fixing it.

**Comment by platinumazure** [comment]
Thanks for the issue!

The enforcement of space (or lack thereof) between the function parens and curly braces is handled by the [`space-before-blocks`](https://eslint.org/docs/rules/space-before-blocks) rule. Please try enabling that rule.

`keyword-spacing` handles whitespace around the keyword (`function`) itself. For example:

```js
const foo = function () {};
                 // ^-- keyword-spacing checks this space
```

Hope this helps!

**Comment by arimourao** [comment]
Thanks for the fast reply! I think this rule is not what I need. I want a rule to enforce a space between "()"  and the curly braces. ie:
```function() {};
             //^ no space
               |
    if() {};

```

**Comment by platinumazure** [comment]
I'm a bit confused. The words you are using don't quite match the code samples.

Please tell me what you're looking for for each space in the following examples (Always/Never/Don't Care):

```js
function foo () {}
// 1. --^   ^  ^
// 2. ------|  |
// 3. ---------|

/* padding */ if (foo) {}
// 1. ----------^     ^
// 2. ----------------|
```

**Comment by arimourao** [comment]
Using your example, my rule would be: 
function: 1 always, 2 never, 3 always
if: 1 never, 2 always

**Comment by platinumazure** [comment]
@arimourao Thanks, here's what you would want:

```json
{
    "rules": {
        "keyword-spacing": ["error", { "after": "never" }],
        "space-before-blocks": ["error", "always"],
        "space-before-function-paren": ["error", "never"]
    }
}
```

Be sure to look at the [`keyword-spacing` docs](https://eslint.org/docs/rules/keyword-spacing) in case you decide you want to have some spacing after certain keywords (using the `overrides` option). Also, I don't know what you will want for the `"before"` option, so be sure to take a look and specify that in your configuration if needed.

Hope this helps!

**Comment by arimourao** [comment]
I got this error:

Configuration for rule "keyword-spacing" is invalid:
        Value "never" should be boolean.

**Comment by platinumazure** [comment]
My mistake, should be `"never": false`. I've edited my previous comment to reflect that.

**Comment by arimourao** [comment]
Thanks for your time @platinumazure that was exactly what I needed!

**Comment by platinumazure** [comment]
@arimourao You're welcome! I'm going to go ahead and close this issue, but please do comment back if you need this reopened or have a related question.

By the way, you can also ask for help on our [Gitter channel](https://gitter.im/eslint/eslint) in future, if you like.
